examples/audio/soundtrack.rs

```diff
@@ -0,0 +1,157 @@
+        .add_systems(Startup, setup)
+        .add_systems(Update, (cycle_game_state, fade_in, fade_out))
+        .add_systems(Update, change_track)
+        .run();
+}
+
+// This resource simulates game states
+#[derive(Resource, Default)]
+enum GameState {
+    #[default]
+    Peaceful,
+    Battle,
+}
+
+// This timer simulates game state changes
+#[derive(Resource)]
+struct GameStateTimer(Timer);
+
+//  This resource will hold the track list for your soundtrack
+#[derive(Resource)]
+struct SoundtrackPlayer {
+    track_list: Vec<Handle<AudioSource>>,
+}
+
+impl SoundtrackPlayer {
+    fn new(track_list: Vec<Handle<AudioSource>>) -> Self {
+        Self { track_list }
+    }
+}
+
+// This component will be attached to an entity to fade the audio in
+#[derive(Component)]
+struct FadeIn;
+
+// This component will be attached to an entity to fade the audio out
+#[derive(Component)]
+struct FadeOut;
+
+fn setup(asset_server: Res<AssetServer>, mut commands: Commands) {
+    // Instantiate the game state resources
+    commands.insert_resource(GameState::default());
+    commands.insert_resource(GameStateTimer(Timer::from_seconds(
+        10.0,
+        TimerMode::Repeating,
+    )));
+
+    // Create the track list
+    let track_1 = asset_server.load::<AudioSource>("sounds/Mysterious acoustic guitar.ogg");
+    let track_2 = asset_server.load::<AudioSource>("sounds/Epic orchestra music.ogg");
+    let track_list = vec![track_1, track_2];
+    commands.insert_resource(SoundtrackPlayer::new(track_list));
+}
+
+// Every time the GameState resource changes, this system is run to trigger the song change.
+fn change_track(
+    mut commands: Commands,
+    soundtrack_player: Res<SoundtrackPlayer>,
+    soundtrack: Query<Entity, With<AudioSink>>,
+    game_state: Res<GameState>,
+) {
+    if game_state.is_changed() {
+        // Fade out all currently running tracks
+        for track in soundtrack.iter() {
+            commands.entity(track).insert(FadeOut);
+        }
+
+        // Spawn a new `AudioBundle` with the appropriate soundtrack based on
+        // the game state.
+        //
+        // Volume is set to start at zero and is then increased by the fade_in system.
+        match game_state.as_ref() {
+            GameState::Peaceful => {
+                commands.spawn((
+                    AudioBundle {
+                        source: soundtrack_player.track_list.first().unwrap().clone(),
+                        settings: PlaybackSettings {
+                            mode: bevy::audio::PlaybackMode::Loop,
+                            volume: bevy::audio::Volume::ZERO,
+                            ..default()
+                        },
+                    },
+                    FadeIn,
+                ));
+            }
+            GameState::Battle => {
+                commands.spawn((
+                    AudioBundle {
+                        source: soundtrack_player.track_list.get(1).unwrap().clone(),
```

Comment:
I did consider this, but was worried that making it too "static" would not be good for the example, so I decided to go with a `Vec` to show that this is flexible. I almost created a struct to hold not only the track, but a title and volume setting as well since the original issue mentioned configurable volume, but opted against this for simplicity. This is my first example so still not sure as to how in depth they need to be, or how flexible/dynamic they should be. 

I also used `unwrap()` statements instead of doing proper error handling since I wanted to keep the code simple and focused on the subject of the example, and if it cannot get the files, then there's not really any point in running the example and panicking should be fine. I also saw that other examples also uses `unwrap()` statements.

If I do write other examples in the future, I will make sure to keep your feedback in mind. Any other feedback or pointers is also appreciated.